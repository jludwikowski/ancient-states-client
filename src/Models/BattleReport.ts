import Army from "./Army";

export default interface BattleReport {
    id: number,
    turnList: Army[][],
    onDisband(id: number): void
}
