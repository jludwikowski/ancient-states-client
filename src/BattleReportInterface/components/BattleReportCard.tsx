import React from "react";
import BattleReport from "../../Models/BattleReport";
import {BattleTurn} from "./BattleTurn";

interface Props {
    report:BattleReport;
}

export const BattleReportCard: React.FunctionComponent<Props> = props => (
    <div>
        <ul className="list-group">
            {props.report.turnList.map((armiesList, index) => (
                <React.Fragment key={index}>
                    <BattleTurn armiesList={armiesList} />
                </React.Fragment>
            ))}
        </ul>
        <button className='btn btn-outline-secondary' onClick={() => props.report.onDisband(props.report.id)}>
            Disband this report
        </button>
    </div>
)
