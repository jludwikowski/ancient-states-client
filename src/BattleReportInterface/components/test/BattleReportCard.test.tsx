import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import BattleReport from "../../../Models/BattleReport";
import { BattleReportCard } from '../BattleReportCard';

describe('BattleReportCard', () => {

    const battleReport:BattleReport = {
        id: 1,
        turnList: [[{id: 7,
            name: 'New Army',
            description: '',
            status: 'camp',
            orderDate: null,
            targetId: null,
            position: '0,0',
            owner: 1,
            commander: null,
            units: [
                {
                    number: 9802,
                    level: 1,
                    baseUnit: 1,
                }
            ]}]],
        onDisband: (id:number) => {
            console.log('On onDismiss');
        }
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BattleReportCard report={battleReport} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<BattleReportCard report={battleReport} />);
        expect(component).toMatchSnapshot();
    });
});
