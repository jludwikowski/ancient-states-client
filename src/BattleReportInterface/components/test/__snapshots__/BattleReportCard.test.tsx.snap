// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattleReportCard Should match snapshot 1`] = `
<div>
  <ul
    className="list-group"
  >
    <BattleTurn
      armiesList={
        Array [
          Object {
            "commander": null,
            "description": "",
            "id": 7,
            "name": "New Army",
            "orderDate": null,
            "owner": 1,
            "position": "0,0",
            "status": "camp",
            "targetId": null,
            "units": Array [
              Object {
                "baseUnit": 1,
                "level": 1,
                "number": 9802,
              },
            ],
          },
        ]
      }
    />
  </ul>
  <button
    className="btn btn-outline-secondary"
    onClick={[Function]}
  >
    Disband this report
  </button>
</div>
`;
